react-validated-input-component/src/password.jsx

```diff
@@ -1,47 +1,53 @@
-import React from 'react'
+import React from 'react';
 
 class Password extends React.Component {
-constructor(props) {
+
+  constructor(props) {
     super(props);
-    this.state = { password: "", valid: false, error: ""};
-    this.handleInput = this.handleInput.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this);
+    this.state = { password: '', valid: false, error: 'A password is required' };
+    this.handleChange = this.handleChange.bind(this);
   }
 
-  handleInput(event) {
-    event.
-    this.setState({password: event.target.value})
-    console.log(this.state)
-  }
+  handleChange(event) {
+    this.setState({ password: event.target.value }, () => {
+      const passLength = this.state.password.length;
```

Comment:
As was discussed in slack, you shouldn't get a value from state right after calling `setState` it will not be the updated value. This can cause unexpected side effects. Here you can get the actual length of the current password by using `event.target.value.length`